source/common/router/router.cc

```diff
@@ -475,9 +475,7 @@ void Filter::onUpstreamReset(UpstreamResetType type,
   if (upstream_request_) {
     upstream_host = upstream_request_->upstream_host_;
     if (upstream_host) {
-      upstream_host->outlierDetector().putHttpResponseCode(
-          enumToInt(type == UpstreamResetType::Reset ? Http::Code::ServiceUnavailable
-                                                     : timeout_response_code_));
+      upstream_host->outlierDetector().connectFailure();
```

Comment:
tcp_proxy also supports outlier detection. I think that should be modified with these new methods?